connectors/sources/notion.py

```diff
@@ -176,17 +176,27 @@ async def fetch_children_recursively(block):
                     ):  # pyright: ignore
                         yield grandchild
 
-        async for block in self.async_iterate_paginated_api(
-            self._get_client.blocks.children.list, block_id=block_id
-        ):
-            if block.get("type") not in ["child_database", "child_page", "unsupported"]:
-                yield block
-                if block.get("has_children") is True:
-                    async for child in fetch_children_recursively(block):
-                        yield child
-            if block.get("type") == "child_database":
-                async for record in self.query_database(block.get("id")):
-                    yield record
+        try:
+            async for block in self.async_iterate_paginated_api(
+                self._get_client.blocks.children.list, block_id=block_id
+            ):
+                if block.get("type") not in [
+                    "child_database",
+                    "child_page",
+                    "unsupported",
+                ]:
+                    yield block
+                    if block.get("has_children") is True:
+                        async for child in fetch_children_recursively(block):
+                            yield child
+                if block.get("type") == "child_database":
+                    async for record in self.query_database(block.get("id")):
+                        yield record
+        except APIResponseError as error:
+            if error.code == "validation_error":
+                self._logger.warning(
+                    f"Skipping children of block with id: {block_id} as not supported by API : {error}"
+                )
```

Comment:
I really don't like this, because it doesn't "fail fast." If there's a bug and we're making invalid requests, we won't catch it unless the customer is watching their logs. Instead, I'd rather raise errors _unless_ we know exactly what caused this error and can justify ignoring it other than logging that it happened.

Reply:
What was the API response for the block, before we tried to fetch its children? There's got to be ways to identify that we shouldn't try to fetch the children of these.

Reply:
Hey, the file [block-response.json](https://github.com/elastic/connectors/files/14947924/block-response.json), is the response for the block before we try to fetch its children.

Reply:
🤦 I misunderstood. 
Ok, yeah, there's not a lot of info there.

Is the error more informative? Maybe I'd be less hesitant to swallow this error if we could be really narrow in what we moved past. Like, is the error message informative at all, and something that we could match on only if it's this specific error case of an external object?

Reply:
The error is not specific to external_objects. Following is the response we get:
```
{
    "object": "error",
    "status": 400,
    "code": "validation_error",
    "message": "Block type external_object_instance_page is not supported via the API.",
    "developer_survey": "https://notionup.typeform.com/to/bllBsoI4?utm_source=postman",
    "request_id": "9f987592-c194-4cfd-afa7-0397f22faee6"
}
```

Reply:
That looks very specific to me. Like we could do:

```
try:
  do_thing()
except APIResponseError as error:
  if error.code == "validation_error" and "external_object" in error.message:
    log.warn("Encountered external object. Skipping")
  else:
    raise
```

The `and "external_object" in error.message:` would hopefully ensure we only do this for this narrow subset of "external_object" issues that we can't anticipate, but don't hide other bugs that might surface as Validation Errors.

Reply:
Got it, updated with commit id:  [b18f0d4](https://github.com/elastic/connectors/pull/2337/commits/b18f0d4c1ed5b2568ae32c162d183b9942889e1e)